pycomlink/processing/nearby_rain_retrival.py

```diff
@@ -132,10 +132,9 @@ def nearby_rainfall_retrival(
     F_values_correction=True
     """
 
-    # Make sure that we only continue if a correct combination of optional args is used
-    if (f_GHz is not None) and (pol is not None) and (a is None) and (
-            b is None):
-
+    # Make sure that we only continue if a correct combination of optional
+    # args is used
+    if (f_GHz is not None) and (pol is not None) and (a is None) and (b is None):
         if type(f_GHz) is not np.ndarray:
             f_GHz = f_GHz.values
```

Comment:
According to the docstring `f_GHz` should be a `float`. But if I supply a float I guess this code will fail because a float does not have an attribute `.values`. Maybe it would be better here to use `np.atleast_1d` to assure that you always get a numpy array, for providing a float and np.array and even an xr.DataArray (I just checked an np.atleast_1d digest xr.DataArray).

Reply:
Looking at the code below (L148 and following) It might be even better the update our function `a_b` so that I can digest arrays or iterables. Not sure if it does that correctly...

If you think this is a good option, I would still suggest to keep your approach here and open an issue to document the idea of updating the `a_b` function. This issue should then also hold the info that the for-loop in this function can be removed once `a_b` digests arrays and/or iterables appropriately.

Reply:
I have opened issue #140 on this topic.

Reply:
Okay. Good to track that in a new issue.

But what should be changed here is that `f_GHz.values` and `pol.values` will fail if they are passed as suggested in the docstring as float and string, respectively. You either have to update the docstring so that people pass either a numpy.array or a xarray.DataArray. Or you implement my suggested solution using `np.atleast_1d`.

Reply:
Individual float or int values are digested by the function, however a list of floats or ints is not. Therefore, I removed both from the docstring.